code/game/objects/items/books/_book.dm

```diff
@@ -133,9 +133,11 @@
 /obj/item/book/proc/formatpencode(var/mob/user, var/t, var/obj/item/pen/P)
 	. = t
 	if(findtext(t, "\[sign\]"))
-		. = replacetext(t, "\[sign\]", "<font face=\"[signfont]\"><i>[P ? P.get_signature(user) : "Anonymous"]</i></font>")
+		var/decl/tool_archetype/pen/parch = GET_DECL(TOOL_PEN)
+		var/signature = parch.get_signature(user, P) 
+		. = replacetext(t, "\[sign\]", "<font face=\"[signfont]\"><i>[signature]</i></font>")
 	if(P)
-		if(P.iscrayon)
+		if(P.get_tool_property(TOOL_PEN, TOOL_PROP_PEN_FLAG) & PEN_FLAG_CRAYON)
```

Comment:
Probably want to retrieve these property values and keep them rather than retrieving them repeatedly in this chain.

Reply:
True!